yb-voyager/cmd/cutover.go

```diff
@@ -51,20 +51,24 @@ func InitiatePrimarySwitch(action string) error {
 }
 
 func createTriggerIfNotExists(triggerName string) error {
-	triggerFPath := filepath.Join(exportDir, "metainfo", "triggers", triggerName)
-	if utils.FileOrFolderExists(triggerFPath) {
+	msr, err := metaDB.GetMigrationStatusRecord()
+	if err != nil {
+		log.Errorf("creating trigger(%s): %v", triggerName, err)
+		return fmt.Errorf("creating trigger(%s): %w", triggerName, err)
+	}
+
+	if msr != nil && msr.CheckIfTriggerExists(triggerName) {
 		utils.PrintAndLog("%s already initiated, wait for it to complete", triggerName)
 		return nil
 	}
```

Comment:
You can perform this check right in the closure that you pass to the `UpdateMigrationStatusRecord()`. The method already reads the msr and passes it to the closure. You can inspect the object as you need.

Reply:
done